taf/tests/test_updater.py

```diff
@@ -397,6 +400,29 @@ def test_update_repo_wrong_flag(updater_repositories, origin_dir, client_dir):
     )
 
 
+def test_update_repo_target_in_indeterminate_state(
+    updater_repositories, origin_dir, client_dir
+):
+    repositories = updater_repositories[
+        "test-updater-target-repository-has-indeterminate-state"
+    ]
+    origin_dir = origin_dir / "test-updater-target-repository-has-indeterminate-state"
+
+    targets_repo_path = client_dir / TARGET_REPO_REL_PATH
+
+    _update_and_check_commit_shas(
+        None, repositories, origin_dir, client_dir, UpdateType.OFFICIAL
+    )
+    # Create an `index.lock` file, indicating that an incomplete git operation took place
+    # index.lock is created by git when a git operation is interrupted.
+    with open(str(Path(targets_repo_path, ".git", "index.lock")), "w"):
```

Comment:
It's a bit cleaner to use pathlib's touch here

Reply:
did not know that, awesome.